ch_pipeline/hfb/analysis.py

```diff
@@ -1047,3 +1048,59 @@ def _ensure_list(x):
         y = [x]
 
     return y
+
+
+class HFBMedianSubtraction(task.SingleTask):
+    """Subtracting weighted median along beam axis to remove fluctuations in the
+    data induced by temperature fluctuations in East and West receiver huts."""
+
+    def process(self, stream):
+        """Subtract weighted median of the data along beam axis from the data.
+        A binary mask (0 when the weight is zero, and 1 for non-zero weights)
+        is used in weighted median
+
+        Parameters
+        ----------
+        stream : containers.HFBData
+            Container with HFB data and weights.
+
+        Returns
+        -------
+        out : containers.HFBData
+            Container with HFB data and weights.
+        """
+
+        # Extract data from container
+        data = stream.hfb[:]
+        weight = stream.weight[:]
+
+        # Change data to numpy array, so that it can be reshaped
+        if isinstance(data, mpiarray.MPIArray):
+            data = data.local_array
+            weight = weight.local_array
+
+        # make a mask of non-zero weights
+        mask = weight != 0
+
+        # Generate binary weight
+        binary_weight = mask.astype(float)
```

Comment:
I prefer to explicitly set the precision. In this case `np.float32` would suffice.